### ISSUE TITLE
mouseposition control: add option not to clear out when mouse leaves map viewport

### ISSUE THREAD

**Issue by dasakigr** [openlayers/openlayers#2914]
I am new in github so sorry for my mistakes.
I started to learn ol3 recently. I noticed that when the mouse is removed from the map erases the contents of mouseposition something that does not happen in other programs GIS.
I managed this by changing the function undefinedHTML but there was a problem. There isn't setUndefinedHTML method. This I managed by changing the mousePositionControl.O variable when the mouse moves equal to the mouseposition text each time.
I try to set mousePositionControl.O variable on event mousepositionout but didn't work.
So the scope is to fix setUndefinedHTML and mousepositionout event

**Comment by pedros007** [comment]
Maintainers (@tschaub @bartvde and @ahocevar?) What do you think of one of the following ways to implement this? I could try and work on a pull request.

1. Add an ol.control.MousePosition option to override [handleMouseOut](https://github.com/openlayers/openlayers/blob/c2f7baf940d0cbf5466d05076c5fee006f37379c/src/ol/control/MousePosition.js#L159-L162), similar to the way a user can override the `render` method
2. Add an option to  to not bind  [EventType.MOUSEOUT](https://github.com/openlayers/openlayers/blob/c2f7baf940d0cbf5466d05076c5fee006f37379c/src/ol/control/MousePosition.js#L176-L177)

**Comment by marcjansen** [comment]
I would suggest to add an option `autoClear`, `clearOnMouseOut` (or a similar name) with `true` being the default. A PR would be welcomed!

**Comment by fredj** [comment]
And what if instead of adding a new option, setting `undefinedHTML` to `undefined` keep the value?
It will break the API but it will be logical to me: `undefined` doesn't do anything and an empty string (or any string) replaces the coordinates with it

**Comment by pedros007** [comment]
@marcjansen Sounds good, thanks.  I have it coded, but I am running into trouble writing a test for the feature.

[Here's a CodePen of my attempt](https://codepen.io/pschmitt/pen/WMGpaZ). I set a breakpoint in `MousePosition.prototype.handleMouseMove`, but it never seems to be triggered.  I expected this code to trigger [handleMouseMove](https://github.com/openlayers/openlayers/blob/master/src/ol/control/MousePosition.js#L148-L152):
```
pointerEvent = new ol.pointer.PointerEvent('mousemove', {clientX: 200, clientY: 300})
event = ol.MapBrowserPointerEvent('mousemove', map, pointerEvent);
map.handleMapBrowserEvent(event);
```
Can you offer any guidance?

**Comment by pedros007** [comment]
I've figured out how to trigger the event from my test. I'll wrap up & submit a pull request soon.

Code which would not trigger [MousePosition.prototype.handleMouseMove](https://github.com/openlayers/openlayers/blob/master/src/ol/control/MousePosition.js#L148-L152):
```javascript
      const viewport = map.getViewport();
      const position = viewport.getBoundingClientRect();
      const pointerEvent = new PointerEvent(type, {
        clientX: position.left + x + width / 2,
        clientY: position.top + y + height / 2
      });
      const event = new MapBrowserPointerEvent(type, map, pointerEvent);
```
with this: 
```javascript
      const viewport = map.getViewport();
      var evt = new MouseEvent('mousemove', {
        clientX:  position.left + x + width / 2,
        clientY:  position.top + y + height / 2
      });
      document.getElementsByClassName('ol-viewport')[0].dispatchEvent(evt);
```

**Comment by pedros007** [comment]
@marcjansen PR #7784 is ready.

**Comment by pedros007** [comment]
This issue can be closed now that #7784 has been merged. With OpenLayers > 4.6.5, Mouse position will be retained when the mouse leaves the viewport and the `undefinedHTML` option is set to `undefined`, like so:
```javascript
new ol.control.MousePosition({ undefinedHTML: undefined })
```

**Comment by pedros007** [comment]
See #8109, which has a nicer summary of the options: 

> So this is how the undefinedHTML option now works:
> 
> * by default, mouseout will clear the previous mouse position by rendering '&nbsp;'
> * if you want something else rendered on mouseout, provide a string (e.g. 'nothing to see here')
> * if you want the mouse position to be retained on mouseout, provide a falsey value (e.g. '')
